src/components/map/map.tsx

```diff
@@ -226,6 +245,22 @@ export class Map implements GxComponent {
     }
   }
 
+  private onMapCircleDeleted(pCircle: circle) {
+    let i = 0;
+    pCircle.remove();
+    while (
+      i <= this.circleList.length &&
+      this.circleList[i]._leaflet_id !== pCircle._leaflet_id
+    ) {
+      i++;
+    }
+    if (i <= this.circleList.length) {
+      this.circleList.splice(i, 1);
+    } else {
+      console.warn("There was an error in the circle list!");
+    }
+  }
+
```

Comment:
It would be great if we could reuse this logic for the different types of shapes so that we can avoid code repetition. What do you think?
Feel free to address this in a future PR